darwin/future/helpers/exception_handler.py

```diff
@@ -0,0 +1,37 @@
+import logging
+import sys
+from typing import List, Optional, Union
+
+from rich.console import Console
+
+
+def handle_exception(exception: Optional[Union[Exception, List[Exception]]]) -> None:
+    """
+    Handles an exception or list of exceptions by printing them to the terminal
+
+    Parameters
+    ----------
+    exception : Optional[Union[Exception, List[Exception]]]
+        The exception(s) to handle
+    """
+    IS_INTERACTIVE_SESSION = sys.stdout and sys.stdout.isatty()
+
+    if not exception:
+        exc_info = sys.exc_info()
+        if exception := getattr(exc_info, "[1]", None):
+            ...
+        else:
+            raise ValueError("No exception provided and no exception in sys.exc_info")
+
+    if IS_INTERACTIVE_SESSION:
+        console = Console()
+        handler = console.print
+    else:
+        logger = logging.getLogger(__name__)
```

Comment:
We should definitely use more logging in our code, but we also have to define a good strategy of where, what to capture. I think our current strategy just captures the CLI command being run which is kind of useless as the customer can just tell us that anyway.

Reply:
Yeah, agreed, but it feels like a piecemeal job like this, rather than a big all in one job?